src/datasets/arrow_dataset.py

```diff
@@ -4140,6 +4146,38 @@ def push_to_hub(
         info_to_dump.splits = {
             split: SplitInfo(split, num_bytes=dataset_nbytes, num_examples=len(self), dataset_name=dataset_name)
         }
+        if config.DATASETDICT_INFOS_FILENAME in repo_files:
```

Comment:
If the dataset_infos.json file already exists, then it seems that the the previous `info_to_dump` defined a few lines above is overwritten

Maybe move the code that creates `info_to_dump` when there's no `datasets_infos.json` in a `else` statement after this `if` ?

Reply:
There is an additional nested `if` inside this `if` block, so we need two identical `else` blocks (one for this `if` and one for the nested `if`). Let me know which style you prefer.

Reply:
Oh indeed, makes sense !